docs/proposals/ingesters-migration.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "Migrating ingesters from chunks to blocks and back."
+linkTitle: "Migrating ingesters from chunks to blocks and back."
+weight: 1
+slug: ingesters-migration
+---
+
+- Author: @pstibrany
+- Reviewers:
+- Date: June 2020
+- Status: Proposed
+
+# Migrating ingesters from chunks to blocks
+
+This short document describes the first step in full migration of the Cortex cluster from using chunks storage to using blocks storage, specifically switching ingesters to using blocks, and modification of queriers to query both chunks and blocks storage.
+
+## Ingesters
+
+When switching ingesters from chunks to blocks, we need to consider the following:
+- Ingesting of new data, and querying should work during the switch.
+- Ingesters are rolled out with new configuration over time. There is overlap: ingesters of both kinds (chunks, blocks) are running at the same time.
+- Ingesters using WAL (all environments in GL) don’t flush in-memory chunks to storage on shutdown.
+- Rollout should be as automated as possible.
+
+How do we handle ingesters with WAL? There are several possibilities, but the simplest option seems to be adding a new flag to ingesters to flush chunks on shutdown. This is trivial change to ingester, and allows us to do automated migration by:
+1) enabling this flag on each ingester (first rollout).
+2) turn off chunks, enable TSDB (second rollout). During the second rollout, as the ingester shuts down, it will flush all chunks in memory, and when it restarts, it will start using TSDB.
```

Comment:
Just to fix the website markdown compilation.
```suggestion

1. Enabling this flag on each ingester (first rollout).
2. Turn off chunks, enable TSDB (second rollout). During the second rollout, as the ingester shuts down, it will flush all chunks in memory, and when it restarts, it will start using TSDB.
```